src/EventService.Business/Commands/Event/CreateEventCommand.cs

```diff
@@ -0,0 +1,147 @@
+    _eventCategoryRepository = eventCategoryRepository;
+    _eventMapper = eventMapper;
+    _eventUserMapper = eventUserMapper;
+    _eventCategoryMapper = eventCategoryMapper;
+    _validator = validator;
+    _accessValidator = accessValidator;
+    _responseCreator = responseCreator;
+    _contextAccessor = contextAccessor;
+    _userService = userService;
+    _emailService = emailService;
+  }
+
+  private async Task SendInviteEmailsAsync(List<Guid> userIds, string eventName)
+  {
+    List<UserData> usersData = await _userService.GetUsersDataAsync(userIds);
+
+    if (usersData is null || !usersData.Any())
+    {
+      return;
+    }
+
+    foreach (UserData user in usersData)
+    {
+      await _emailService.SendAsync(
+        user.Email,
+        "Invite to event",
+        $"You have been invited to event {eventName}");
+    }
+  }
+
+  public async Task<OperationResultResponse<bool>> ExecuteAsync(CreateEventRequest request)
+  {
+    Guid senderId = _contextAccessor.HttpContext.GetUserId();
+    bool hasAddEditRemoveUsersRights = await _accessValidator.HasRightsAsync(senderId, Rights.AddEditRemoveUsers);
+    OperationResultResponse<bool> response = new();
+
+    if (!hasAddEditRemoveUsersRights)
+    {
+      return _responseCreator.CreateFailureResponse<bool>(HttpStatusCode.Forbidden);
+    }
+
+    if (request.Users.Distinct().Count() != request.Users.Count())
+    {
+      response.Errors = new List<string>() { "Some duplicate users have been removed from the list." };
+      request.Users = request.Users.Distinct().ToList();
+    }
+
+    request.CategoryIds = request.CategoryIds.Distinct().ToList();
+
+    ValidationResult validationResult = await _validator.ValidateAsync(request);
+    if (!validationResult.IsValid)
+    {
+      return _responseCreator.CreateFailureResponse<bool>(
+        HttpStatusCode.BadRequest,
+        validationResult.Errors.Select(er => er.ErrorMessage).ToList());
+    };
+
+    DbEvent dbEvent = _eventMapper.Map(request, senderId);
+
+    if ((dbEvent.Access == AccessType.Closed && !hasAddEditRemoveUsersRights)
+        || !(dbEvent.Access == AccessType.Opened
+          && (hasAddEditRemoveUsersRights
+            || (!hasAddEditRemoveUsersRights
+              && request.Users.Count == 1
+              && request.Users.Exists(x => x.UserId == senderId)))))
+    {
+      return _responseCreator.CreateFailureResponse<bool>(HttpStatusCode.Forbidden);
+    }
+
+    response.Body = await _eventRepository.CreateAsync(dbEvent);
+
+    List<DbEventUser> dbEventUsers = _eventUserMapper.Map(request, senderId, dbEvent.Id);
+    response.Body = await _eventUserRepository.CreateAsync(dbEventUsers) 
+      && response.Body;
+
+    await SendInviteEmailsAsync(dbEventUsers.Where(x => x.Status == EventUserStatus.Invited).Select(x => x.UserId).ToList(), dbEvent.Name);
+
+    response.Body = await _eventCategoryRepository.CreateAsync(_eventCategoryMapper.Map(request, senderId, dbEvent.Id))
+      && response.Body;
```

Comment:
try to call mapper in command one time (than in DbEventMapper call mappers for users and categories);
and try to call repository one time (dbEvent can have inside users and categories) -> one DB call + no && response.Body here

Reply:
Done but with three calls to DB because I have to insert data in three tables (Event, EventUsers, EventCategories)